### PR TITLE
C unwind is stable

### LINKED ISSUE #2581: The `c_unwind` ABI is stable in 1.71
Now that [`c_unwind` has been stabilized](https://blog.rust-lang.org/2023/07/13/Rust-1.71.0.html#c-unwind-abi) we can put it under the right Rust version instead of just using nightly.

### PR DESCRIPTION
**Pull request by tshepang**
Closes #2581

### CODE REVIEW

**Review by pvdrz — CHANGES_REQUESTED** [review]
Could you update the `bindgen-tests/tests/expectations/tests/c-unwind-abi-override-nightly.rs and `bindgen-tests/tests/expectations/tests/c-unwind-abi-override.rs` tests so we can test this?

**Comment by tshepang** [comment]
Did changes, but not clear why `abi_thiscall` feature was part of the test (so I removed it).

Am also not happy with my changelog entry... happy to hear suggestions.

**Comment by pvdrz** [comment]
> Did changes, but not clear why `abi_thiscall` feature was part of the test (so I removed it).
I suspect some dummy forgot to change the feature when they copied the tests for c_unwind (I'm 99% sure I'm that dummy)
 
> Am also not happy with my changelog entry... happy to hear suggestions.
Maybe something like: The `c_unwind` ABI can be used without a feature gate for any rust target version greater or equal to 1.71

File: bindgen/features.rs [hunk 1]
```diff
@@ -306,8 +310,7 @@ mod test {
                 f_nightly.maybe_uninit &&
                 f_nightly.repr_align &&
                 f_nightly.thiscall_abi &&
-                f_nightly.vectorcall_abi &&
-                f_nightly.c_unwind_abi
+                f_nightly.vectorcall_abi
```
Comment (pvdrz) [inline bindgen/features.rs:313]: Maybe add this:
```rust
let f_1_68 = RustFeatures::from(RustTarget::Stable_1_68);
assert!(f_1_68.abi_efiapi && !f_1_68.c_unwind_abi);
```

**Review by pvdrz — APPROVED** [review]

**Pull request merged**